cg/test/processors/processors.listeners.spec.js

```diff
@@ -162,4 +164,40 @@ describe('Processors listening for blockchain events', () => {
       done();
     }, 4000);
   });
+
+  it('Should react to restriction events', async done => {
+    // Given
+    // We need to insert a subject this way because the mocks override the endpoints (give-consent in this case)
+    let subjectId = sha3('58669879');
+    const encryptionKey = encryption.generateClientKey();
+    await db('subjects').insert({
+      id: subjectId,
+      personal_data: encryptForStorage(JSON.stringify({ data: 'some data' }), encryptionKey),
+      direct_marketing: true,
+      email_communication: true,
+      research: true
+    });
+    await db('subject_keys').insert({
+      subject_id: subjectId,
+      key: encryptionKey
+    });
+
+    // When
+    await recordRestrictionByController(subjectId, false, true, false);
+
+    // Then
+    setTimeout(async () => {
+      // We have to wait for the processor to update his database with the new subject restrictions
```

Comment:
is there a way to force a refresh?

Reply:
It's listening to an event, I can't see another way to refresh the restrictions. There might be, but I can't see

Reply:
ok thank you, it was just for my own information